File: engine/engine-model/src/main/scala/tools/packet/cashshop/operation/ShowCashShopMessage.scala
package tools.packet.cashshop.operation

import net.opcodes.SendOpcode
import tools.packet.PacketInput
import tools.packet.cashshop.CashShopMessage

class ShowCashShopMessage(private var _message: CashShopMessage) extends PacketInput {
  def message: CashShopMessage = _message

  override def opcode(): SendOpcode = SendOpcode.CASH_SHOP_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/cashshop/operation/ShowCashShopMessage.java
package tools.packet.cashshop.operation;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;
import tools.packet.cashshop.CashShopMessage;

public record ShowCashShopMessage(CashShopMessage message) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CASH_SHOP_OPERATION;
   }
}